src/Xrefcheck/Verify.hs

```diff
@@ -137,9 +136,15 @@ data VerifyError
   | ExternalFtpException FTPException
   | FtpEntryDoesNotExist FilePath
   | ExternalResourceSomeError Text
-  | PermanentRedirectError Text (Maybe Text)
+  | RedirectChainCycle (NonEmpty Text)
+  | RedirectChainBroken (NonEmpty Text)
+  | RedirectRuleError (Maybe RedirectRuleOn) (NonEmpty Text) (Maybe Text)
```

Comment:
I still think that at least the last field is worth clarification.

Reply:
It corresponds to the last step in the redirect chain. In fact, I think that it would be better if it is included in the same redirect chain as in the other error constructors.